pyicloud/base.py

```diff
@@ -84,20 +85,21 @@ def request(self, *args, **kwargs):
 
         logger.debug(json)
 
-        reason = json.get('errorMessage')
-        reason = reason or json.get('reason')
-        reason = reason or json.get('errorReason')
-        if not reason and isinstance(json.get('error'), six.string_types):
-            reason = json.get('error')
-        if not reason and json.get('error'):
-            reason = "Unknown reason"
+        if isinstance(json, dict):
```

Comment:
This change seems to be unrelated, why is it needed?

Reply:
For the Drive requests, `json` is not a dict, it's a list, which evidently is not something that had been encountered before. Either the API works differently, or I'm not making the right call?